pylops/basicoperators/MemoizeOperator.py

```diff
@@ -29,11 +33,8 @@ class MemoizeOperator(LinearOperator):
 
     """
 
-    def __init__(self, Op, max_neval=10):
-        self.Op = Op
-        self.shape = Op.shape
-        self.dtype = np.dtype(Op.dtype)
-        self.explicit = False
+    def __init__(self, Op, max_neval=10, name="M"):
+        super().__init__(Op=Op, explicit=False, clinear=None, name=name)
```

Comment:
The memoized operator should probably just inherit the name of the original operator.

Reply:
Agree